test_data/benchmark_main.py

```diff
@@ -12,7 +12,7 @@
 # How much faster Rust should be than other implementations
 RATIOS_SBP2JSON = {
     "haskell": 2.06,
-    "python": 25.32,
+    "python": 18.8,
```

Comment:
@jayvdb sneaking this in, this is failing occasionally now, so there does appear to be a pretty good speed-up

Reply:
Actually had to revert this again, seems on some runs it's going faster, and going outside the 25% range, on others it's not.

Reply:
It was 2.74 before jit was removed, so this much higher number still concerns me a bit.

It guess it means the benchmark we have is not covering a similar load that @dgburr is using, as he saw perf return to almost the jit-era perf.  So perf regressions of a load like @dgburr is using can still silently occur and not trigger a CI failure.

Reply:
> 
> 
> It was 2.74 before jit was removed, so this much higher number still concerns me a bit.
> 
> It guess it means the benchmark we have is not covering a similar load that @dgburr is using, as he saw perf return to almost the jit-era perf. So perf regressions of a load like @dgburr is using can still silently occur and not trigger a CI failure.

I think @dgburr is not unpacking/parsing the messages (which goes through `construct`), that and JSON output are a big source of slowness in the Python sbp2json.  If I disable JSON output and unpacking I see similar speed improvements to what @dgburr was seeing.

Reply:
@silverjam: you are correct - I deliberately designed Drongo around the idea of postponing the `dispatch` step (i.e. when the type-specific fields get unpacked) for as long as possible for this exact reason.  So my profiling tests last week were focused mostly on parsing/deframing rather than dispatching/unpacking.  With my latest improvements to the STEP Log Postprocessing Utility I have had to do a lot of message dispatching much earlier than I did previously which has significantly degraded performance again.  It would be really nice to have JIT-accelerated message unpacking, or failing that, a rust-based parser which produces fully unpacked Python object types.